http.go

```diff
@@ -86,6 +102,58 @@ func (s *LumbermillServer) serveHealth(w http.ResponseWriter, r *http.Request) {
 	w.WriteHeader(http.StatusOK)
 }
 
+func (s *LumbermillServer) serveInfluxDBHealth(w http.ResponseWriter, r *http.Request) {
+	errorCount := 0
+
+	// Copy the map so we can unlock.
+	s.recentTokensLock.RLock()
+	tokenMap := make(map[string]string)
+	for name, token := range s.recentTokens {
+		tokenMap[name] = token
+	}
+	s.recentTokensLock.RUnlock()
+
+	for name, token := range tokenMap {
+		// Ugh. For now. Should reuse connections, but tied up in Posters, and abstraction.
+		clientConfig := createInfluxDBClient(name, os.Getenv("INFLUXDB_SKIP_VERIFY") == "true")
+		client, _ := influx.NewClient(&clientConfig)
+
+		// Query the last point for the token and ensure it's been published to in the last XXX minutes.
+		for _, prefix := range influxDbSeriesCheckPrefixes {
+			series := prefix + token
+			query := fmt.Sprintf("select * from %s limit 1", series)
+
+			results, err := client.Query(query, influx.Second)
+			if err != nil || len(results) == 0 {
+				errorCount++
+				log.Printf("at=influxdb-health err=%q result_length=%d host=%q token=%q", err, len(results), name, token)
+			} else {
+				t, ok := results[0].Points[0][0].(float64)
+				if !ok {
+					errorCount++
+					log.Printf("at=influxdb-health err=\"time column was not an int\" host=%q token=%q", name, token)
+					continue
+				}
+
+				ts := time.Unix(int64(t), int64(0)).UTC()
+				now := time.Now().UTC()
+				if now.Sub(ts) > influxDbStaleTimeout {
+					errorCount++
+					log.Printf("at=influxdb-health err=\"stale data\" host=%q ts=%q now=%q token=%q", name, ts, now, token)
+				}
+			}
+		}
+	}
+
+	if errorCount > 0 {
+		log.Printf("at=influxdb-health err=\"non-zero error count during health check\" count=%d", errorCount)
+		http.Error(w, "Failed Health Check", 500)
```

Comment:
The information is in the logs, but I guess that's not very accessible. I'll save off the hosts and return them. I guess since the ELB isn't hitting this, ServiceUnavailable will be ok...